src/backend/gpopt/utils/CMemoryPoolPalloc.cpp

```diff
@@ -47,11 +47,6 @@ CMemoryPoolPalloc::NewImpl(const ULONG bytes, const CHAR *, const ULONG,
 
 		void *ptr = gpdb::GPDBMemoryContextAlloc(m_cxt, alloc_size);
 
-		if (NULL == ptr)
-		{
-			return NULL;
-		}
-
 		SArrayAllocHeader *header = static_cast<SArrayAllocHeader *>(ptr);
 
 		header->m_user_size = bytes;
```

Comment:
Seems here would be a `SIGSEGV`, if ptr is `NULL`. Maybe it's worth to add `GPOS_OOM_CHECK(ptr)` like at `CMemoryPoolTracker`?

Reply:
> Seems here would be a `NULL`-pointer dereference, if ptr is `NULL`.

No! GPDBMemoryContextAlloc throws exception in this case!

Reply:
Sorry, I've forgot to submit review, so this message

>> Or GPDBMemoryContextAlloc will raise an error? If so, maybe it's worth to add comment about that

wasn't visible